### PR TITLE
Django Setup

### PR DESCRIPTION
**Pull request by skthinks**

### CODE REVIEW

**Comment by vishesh92** [comment]
Use better branch names.

File: slacksite/slackbot/views.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    ans = check_present(userid)
+    if check_present(userid):
+        return
+    new_user = User(user_name = username, user_id = userid)
+    new_user.save()
+    user_list = User.objects.all()
+    for user in user_list:
+        print user.user_name
+
+# def get_users_db():
```
Comment (vishesh92) [inline slacksite/slackbot/views.py]: Commented code.

File: slacksite/slackbot/views.py [hunk 2]
```diff
@@ -0,0 +1,42 @@
+from django.shortcuts import render
+
+# Create your views here.
+import psycopg2
```
Comment (vishesh92) [inline slacksite/slackbot/views.py]: `psycopg2` imported but never used.

File: slacksite/slackbot/views.py [hunk 3]
```diff
@@ -17,26 +12,53 @@ def check_present(user_id):
     return False
 
 
+def check_present_stats(user_id):
```
Comment (vishesh92) [inline slacksite/slackbot/views.py]: These type of functions should not be present in views.

File: slacksite/slackbot/views.py [hunk 4]
```diff
@@ -17,26 +12,53 @@ def check_present(user_id):
...
         return
-    new_user = User(user_name = username, user_id = userid)
+    new_user = User(
+        user_name = username, 
+        user_id = userid
+    )
     new_user.save()
+
+
+def insert_stats_db(req):
```
Comment (vishesh92) [inline slacksite/slackbot/views.py]: Choose better function names.

File: slacksite/slackbot/views.py [hunk 5]
```diff
@@ -17,26 +12,53 @@ def check_present(user_id):
...
+    message = req.get('text')
+    if not check_present_stats(userid):
+        new_entry = Leaderboards(
+            user_id = userid, 
+            leave_early = 0,  
+            arrive_late = 0,
+            on_leave = 0
+        )
+        new_entry.save()
+    # get_stats_db()
```
Comment (vishesh92) [inline slacksite/slackbot/views.py]: Remove comments like these.

File: slacksite/slackbot/models.py [hunk 6]
```diff
@@ -8,3 +8,8 @@ class User(models.Model):
     user_id = models.CharField(max_length=30)
 
 
+class Leaderboards(models.Model):
```
Comment (vishesh92) [inline slacksite/slackbot/models.py]: The schema for this should be something like this.

```
date
user_id -> foreign key
count
type
```

File: slacksite/slackbot/urls.py [hunk 7]
```diff
@@ -0,0 +1,7 @@
+from django.conf.urls import url
+
+from . import views
+
+urlpatterns = [
+    url(r'', views.posting, name='Posting'),
```
Comment (vishesh92) [inline slacksite/slackbot/urls.py]: Use this regex, `^$`

File: slacksite/slackbot/views.py [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+
+
+def check_present_stats(user_id):
+    user_id_list = Leaderboards.objects.values_list('user_id', flat=True)
+    if user_id in user_id_list:
+        return True
+    return False
+
+
+def insert_user_db(req):
```
Comment (vishesh92) [inline slacksite/slackbot/views.py]: Use better names.

File: slacksite/slackbot/views.py [hunk 9]
```diff
@@ -0,0 +1,64 @@
+
+from django.http import HttpResponse
+from django.views.decorators.csrf import csrf_exempt
+
+from models import User, Leaderboards
+
+
+def check_present(user_id):
+    user_id_list = User.objects.values_list('user_id', flat=True)
```
Comment (vishesh92) [inline slacksite/slackbot/views.py]: Use this `User.objects.filter(user_id=<user_id>).exists()`

File: slacksite/slackbot/views.py [hunk 10]
```diff
@@ -0,0 +1,64 @@
...
+    if user_id in user_id_list:
+        return True
+    return False
+
+
+def check_present_stats(user_id):
```
Comment (vishesh92) [inline slacksite/slackbot/views.py]: This shouldn't be required with the new schema

File: slacksite/slackbot/views.py [hunk 11]
```diff
@@ -0,0 +1,64 @@
...
+        return
+    new_user = User(
+        user_name = username, 
+        user_id = userid
+    )
+    new_user.save()
+
+
+def insert_stats_db(req):
+    userid = req.get('user_id')
```
Comment (vishesh92) [inline slacksite/slackbot/views.py]: `user_id`

File: slacksite/slackbot/views.py [hunk 12]
```diff
@@ -0,0 +1,64 @@
...
+    if check_present(userid):
+        return
+    new_user = User(
+        user_name = username, 
+        user_id = userid
+    )
+    new_user.save()
+
+
+def insert_stats_db(req):
```
Comment (vishesh92) [inline slacksite/slackbot/views.py]: Choose better name.

File: slacksite/slackbot/views.py [hunk 13]
```diff
@@ -0,0 +1,64 @@
...
+    if not check_present_stats(userid):
+        new_entry = Leaderboards(
+            user_id = userid, 
+            leave_early = 0,  
+            arrive_late = 0,
+            on_leave = 0
+        )
+        new_entry.save()
+    # get_stats_db()
+    print message
```
Comment (vishesh92) [inline slacksite/slackbot/views.py]: Remove `print` statements.

File: slacksite/slackbot/views.py [hunk 14]
```diff
@@ -0,0 +1,64 @@
...
+def get_users_db():
+    user_list = User.objects.all()
+    for user in user_list:
+        print user.user_name
+
+# def get_stats
+
+
+@csrf_exempt
+def posting(request):
```
Comment (dash1291) [inline slacksite/slackbot/views.py:9]: rename

File: slacksite/slackbot/db_utils.py [hunk 15]
```diff
@@ -0,0 +1,24 @@
...
+    new_user.save()
+
+
+def update_offence_db(user_id, message):
+    message = message.lower()
+    offence = ""
+    if "leaving" in message:
+        offence = "leaving_early"
+    elif ("reaching" in message 
+            or "in by" in message
```
Comment (dash1291) [inline slacksite/slackbot/db_utils.py]: better rewrite this as -

```
elif ("reaching" in message or
        "in by" in message or
        "be in" in message):
```

Comment (dash1291) [inline slacksite/slackbot/db_utils.py]: nevermind

**Pull request merged**